spring-kafka/src/test/java/org/springframework/kafka/retrytopic/AsyncMonoRetryTopicScenarioTests.java

```diff
@@ -0,0 +1,1241 @@
+											latchContainer.dltCountdownLatch5,
+											latchContainer.extraDltCountdownLatch5);
+		}
+
+		@Bean
+		MyCustomDltProcessor myCustomDltProcessor6(
+				KafkaTemplate<String, String> kafkaTemplate,
+				CountDownLatchContainer latchContainer) {
+			return new MyCustomDltProcessor(kafkaTemplate,
+											latchContainer.dltCountdownLatch6,
+											latchContainer.extraDltCountdownLatch6);
+		}
+
+	}
+
+	@Configuration
+	static class KafkaProducerConfig {
+
+		@Autowired
+		EmbeddedKafkaBroker broker;
+
+		@Bean
+		ProducerFactory<String, String> producerFactory() {
+			Map<String, Object> props = KafkaTestUtils.producerProps(
+					this.broker.getBrokersAsString());
+			props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+			return new DefaultKafkaProducerFactory<>(props);
+		}
+
+		@Bean("customKafkaTemplate")
+		KafkaTemplate<String, String> kafkaTemplate() {
+			return new KafkaTemplate<>(producerFactory());
+		}
+	}
+
+	@EnableKafka
+	@Configuration
+	static class KafkaConsumerConfig {
+
+		@Autowired
+		EmbeddedKafkaBroker broker;
+
+		@Bean
+		KafkaAdmin kafkaAdmin() {
+			Map<String, Object> configs = new HashMap<>();
+			configs.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, this.broker.getBrokersAsString());
+			return new KafkaAdmin(configs);
+		}
+
+		@Bean
+		ConsumerFactory<String, String> consumerFactory() {
+			Map<String, Object> props = KafkaTestUtils.consumerProps(
+					this.broker.getBrokersAsString(),
+					"groupId",
+					"false");
+			return new DefaultKafkaConsumerFactory<>(props);
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<String, String> retryTopicListenerContainerFactory(
+				ConsumerFactory<String, String> consumerFactory) {
+
+			ConcurrentKafkaListenerContainerFactory<String, String> factory = new ConcurrentKafkaListenerContainerFactory<>();
+			ContainerProperties props = factory.getContainerProperties();
+			props.setIdleEventInterval(100L);
+			props.setPollTimeout(50L);
+			props.setIdlePartitionEventInterval(100L);
+			factory.setConsumerFactory(consumerFactory);
+			factory.setConcurrency(1);
+			factory.setContainerCustomizer(
+					container -> container.getContainerProperties().setIdlePartitionEventInterval(100L));
+			return factory;
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<String, String> kafkaListenerContainerFactory(
+				ConsumerFactory<String, String> consumerFactory) {
+
+			ConcurrentKafkaListenerContainerFactory<String, String> factory = new ConcurrentKafkaListenerContainerFactory<>();
+			factory.setConsumerFactory(consumerFactory);
+			factory.setConcurrency(1);
+			factory.setContainerCustomizer(container -> {
+				if (container.getListenerId().startsWith("manual")) {
+					container.getContainerProperties().setAckMode(AckMode.MANUAL);
+					container.getContainerProperties().setAsyncAcks(true);
+				}
+			});
+			return factory;
+		}
+
+		@Bean
+		TaskScheduler sched() {
```

Comment:
Why do we need scheduler?
Is that something what retry topic relies on?

Reply:
`TaskScheduler` need for `RetryTopicConfiguration`! 
https://github.com/spring-projects/spring-kafka/blob/6f4adb2146a7d27416fdb7810bd0771eeb1022b1/spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationSupport.java#L310-L333